Effy update switch UI

effyyzhang opened the pull request:
- update project name, side panel header font size 
- update header change size on scroll issue 
- update code editor padding 
- update code/preview switch style

nichochar commented:
Ideally screenshots in the description for UI changes please!

benjreinhart commented on packages/web/src/components/apps/workspace/editor/editor.tsx:
```diff
@@ -22,31 +23,24 @@ export function Editor(props: PropsType) {
+          </div>
+        ) : null}
 
-      {/*
```
Can we keep this comment

nichochar commented on packages/web/src/components/apps/workspace/editor/editor.tsx:
```diff
@@ -60,17 +54,42 @@ function CodeEditor({
   const languageExtension = getCodeMirrorLanguageExtension(file);
   const extensions = languageExtension ? [languageExtension] : [];
 
+  useEffect(() => {
```
I'm not sure we want to use refs and an observable pattern here. Concerned this will introduce some jitter.
Can we remove this stuff from the PR? We need to take a pass at looking at options for resizing the code editor but I am suspecting this is not trivial.

nichochar commented on packages/web/src/index.css:
```diff
@@ -156,37 +156,37 @@
     font-size: 3rem;
     line-height: 1;
-    font-weight: 700;
+    font-weight: 500;
```
These styles apply all across the app. Just double checking that:
1. this is intentional
2. you've done a thorough QA of the whole app to make sure it doesn't look weird in some places

effyyzhang replied:
updated, added a screenshot below

effyyzhang commented:
![image](https://github.com/user-attachments/assets/62fe7937-1d76-4c1d-91f9-66f051accad2)
- Update the font weight 
![image](https://github.com/user-attachments/assets/65cc0794-a155-4fdf-be6d-80661ae384f5)
- updated the project header, side panel header font-size, weight
- updated the code/preview switch style 
![header update](https://github.com/user-attachments/assets/34e58655-bccb-4293-9812-5e0ddca1ab46)
- update the header resize move on scroll issue
- update the code editor padding
